examples/substrait-spark/README.md

```diff
@@ -0,0 +1,587 @@
+# Introduction to the Substrait-Spark library
+
+The Substrait-Spark library was recently added to the [substrait-java](https://github.com/substrait-io/substrait-java) project; this library allows Substrait plans to convert to and from Spark Plans.
+
+
+## How does this work in practice?
+
+Once Spark SQL and Spark DataFrame APIs queries have been created, Spark's optimized query plan can be used generate Substrait plans; and Substrait Plans can be executed on a Spark cluster. Below is a description of how to use this library; there are two sample datasets included for demonstration.
+
+The most commonly used logical relations are supported, including those generated from all the TPC-H queries, but there are currently some gaps in support that prevent all the TPC-DS queries from being translatable.
+
+
+## Running the examples
+
+There are 3 example classes:
+
+- [SparkDataset](./app/src/main/java/io/substrait/examples/SparkDataset.java) that creates a plan starting with the Spark Dataset API
+- [SparkSQL](./app/src/main/java/io/substrait/examples/SparkSQL.java) that creates a plan starting with the Spark SQL API
+- [SparkConsumeSubstrait](./app/src/main/java/io/substrait/examples/SparkConsumeSubstrait.java) that loads a Substrait plan and executes it
+
+
+
+### Requirements
+
+To run these you will need:
+
+- Java 17 or greater
+- Docker to start a test Spark Cluster
+  - you could use your own cluster, but would need to adjust file locations defined in [SparkHelper](./app/src/main/java/io/substrait/examples/SparkHelper.java)
+- [just task runner](https://github.com/casey/just#installation) optional, but very helpful to run the bash commands
```

Comment:
minor:
```
the [just](https://github.com/casey/just#installation) task runner
```

Makes it clearer that _just_ is the name of the utility. I definitely read it wrong/fast the first time and thought you meant I just needed a task runner 😅